guppy/src/graph/cargo.rs

```diff
@@ -0,0 +1,365 @@
+    include_dev: bool,
+    host_platform: Option<&'a Platform<'a>>,
+    target_platform: Option<&'a Platform<'a>>,
+    omitted_packages: HashSet<&'a PackageId>,
+}
+
+impl<'a> CargoOptions<'a> {
+    /// Creates a new `CargoOptions` with this resolver version and default settings.
+    ///
+    /// The default settings are similar to what a plain `cargo build` does:
+    ///
+    /// * use version 1 of the Cargo resolver
+    /// * exclude dev-dependencies
+    /// * resolve dependencies assuming any possible host or target platform
+    /// * do not omit any packages.
+    pub fn new() -> Self {
+        Self {
+            version: CargoResolverVersion::V1,
+            include_dev: false,
+            host_platform: None,
+            target_platform: None,
+            omitted_packages: HashSet::new(),
+        }
+    }
+
+    /// Sets the Cargo feature resolver version.
+    ///
+    /// For more about feature resolution, see the documentation for `CargoResolverVersion`.
+    pub fn with_version(mut self, version: CargoResolverVersion) -> Self {
+        self.version = version;
+        self
+    }
+
+    /// If set to true, causes dev-dependencies of the initial set to be followed.
+    ///
+    /// This does not affect transitive dependencies -- for example, a build or dev-dependency's
+    /// further dev-dependencies are never followed.
+    ///
+    /// The default is true, which matches what a plain `cargo build` does.
+    pub fn with_dev_deps(mut self, include_dev: bool) -> Self {
+        self.include_dev = include_dev;
+        self
+    }
+
+    /// Sets both the target and host platforms to the provided one, or to evaluate against any
+    /// platform if `None`.
+    pub fn with_platform(mut self, platform: Option<&'a Platform<'a>>) -> Self {
+        self.target_platform = platform;
+        self.host_platform = platform;
+        self
+    }
+
+    /// Sets the target platform to the provided one, or to evaluate against any platform if `None`.
+    pub fn with_target_platform(mut self, target_platform: Option<&'a Platform<'a>>) -> Self {
+        self.target_platform = target_platform;
+        self
+    }
+
+    /// Sets the host platform to the provided one, or to evaluate against any platform if `None`.
+    pub fn with_host_platform(mut self, host_platform: Option<&'a Platform<'a>>) -> Self {
+        self.host_platform = host_platform;
+        self
+    }
+
+    /// Omits edges into the given packages.
+    ///
+    /// This may be useful in order to figure out what additional dependencies or features a
+    /// particular set of packages pulls in.
+    ///
+    /// This method is additive.
+    pub fn with_omitted_packages(
+        mut self,
+        package_ids: impl IntoIterator<Item = &'a PackageId>,
+    ) -> Self {
+        self.omitted_packages.extend(package_ids);
+        self
+    }
+}
+
+impl<'a> Default for CargoOptions<'a> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```

Comment:
Just for my benefit, if a type has a lifetime param `'a` which isn't used anywhere then what scope is the struct limited by, however long it needs to live? Then if you do set a param that takes a lifetime, then the compiler will shrink it to that lifetime, right?

Reply:
It'll typically be unrestricted aka `'static`, yeah.